crates/cast/bin/cmd/create2.rs

```diff
@@ -167,12 +172,42 @@ mod tests {
 
     #[test]
     fn basic_create2() {
+        // even hex chars
         let args = Create2Args::parse_from(["foundry-cli", "--starts-with", "aa"]);
         let create2_out = args.run().unwrap();
         let address = create2_out.address;
         let address = format!("{address:x}");
 
         assert!(address.starts_with("aa"));
+
+        // odd hex chars
+        let args = Create2Args::parse_from(["foundry-cli", "--starts-with", "aaa"]);
+        let create2_out = args.run().unwrap();
+        let address = create2_out.address;
+        let address = format!("{address:x}");
+
+        assert!(address.starts_with("aaa"));
+
+        // even hex chars with 0x prefix
+        let args = Create2Args::parse_from(["foundry-cli", "--starts-with", "0xaa"]);
+        let create2_out = args.run().unwrap();
+        let address = create2_out.address;
+        let address = format!("{address:x}");
+
+        assert!(address.starts_with("aa"));
+
+        // odd hex chars with 0x prefix
+        let args = Create2Args::parse_from(["foundry-cli", "--starts-with", "0xaaa"]);
+        let create2_out = args.run().unwrap();
+        let address = create2_out.address;
+        let address = format!("{address:x}");
+
+        assert!(address.starts_with("aaa"));
+
+        // TODO: add check fails on wrong chars
+        // let args = Create2Args::parse_from(["foundry-cli", "--starts-with", "0xtest"]);
+        // let create2_out = args.run();
+        // assert_eq!(create2_out, Err("invalid prefix hex provided"));
```

Comment:
I assume this can just be `assert!(create2_out.is_err())`

Reply:
added here https://github.com/foundry-rs/foundry/pull/5714